tests/src/core/geometry/testqgsrectangle.cpp

```diff
@@ -89,6 +89,36 @@ void TestQgsRectangle::fromWkt()
 
   QVERIFY( rect == QgsRectangle::fromWkt( rect.asWktPolygon() ) );
 
+  rect = QgsRectangle::fromWkt( QStringLiteral( "POLYGONZ((0 0 2,1 0 2,1 1 2,0 1 2,0 0 2))" ) );
+  QVERIFY( ! rect.isEmpty() );
+  QCOMPARE( rect.xMinimum(), 0.0 );
+  QCOMPARE( rect.yMinimum(), 0.0 );
+  QCOMPARE( rect.xMaximum(), 1.0 );
+  QCOMPARE( rect.yMaximum(), 1.0 );
+
+  QVERIFY( rect == QgsRectangle::fromWkt( rect.asWktPolygon() ) );
+
+  // this is ok, a single rectangular polygon in a multipolygon object
+  rect = QgsRectangle::fromWkt( QStringLiteral( "MULTIPOLYGON(((0 0,1 0,1 1,0 1,0 0)))" ) );
+  QVERIFY( ! rect.isEmpty() );
+  QCOMPARE( rect.xMinimum(), 0.0 );
+  QCOMPARE( rect.yMinimum(), 0.0 );
+  QCOMPARE( rect.xMaximum(), 1.0 );
+  QCOMPARE( rect.yMaximum(), 1.0 );
+  QVERIFY( rect == QgsRectangle::fromWkt( rect.asWktPolygon() ) );
+
+  // this is ok, a single rectangular polygon in a collection
+  rect = QgsRectangle::fromWkt( QStringLiteral( "GEOMETRYCOLLECTION(MULTIPOLYGON(((0 0,1 0,1 1,0 1,0 0))))" ) );
+  QVERIFY( ! rect.isEmpty() );
+  QCOMPARE( rect.xMinimum(), 0.0 );
+  QCOMPARE( rect.yMinimum(), 0.0 );
+  QCOMPARE( rect.xMaximum(), 1.0 );
+  QCOMPARE( rect.yMaximum(), 1.0 );
+  QVERIFY( rect == QgsRectangle::fromWkt( rect.asWktPolygon() ) );
+
+  QVERIFY( QgsRectangle::fromWkt( QStringLiteral( "xxx" ) ).isEmpty() );
+  QVERIFY( QgsRectangle::fromWkt( QStringLiteral( "LINESTRING ()" ) ).isEmpty() );
+  QVERIFY( QgsRectangle::fromWkt( QStringLiteral( "MULTIPOLYGON()" ) ).isEmpty() );
```

Comment:
I would check for .isNull() here, as .isEmpty() is also true for rectangles containing some information (ie: collapsed vertically and/or horizontally)

Reply:
Done